### PR TITLE
fix: Fix all property, param and return types

### PR DESCRIPTION
**Pull request by provokateurin**
Uses strong types whenever possible, otherwise doc types (or when they add more information). Any other types are removed to inherit from the parents.

### CODE REVIEW

**Comment by provokateurin** [comment]
(I also checked with https://github.com/nextcloud/groupfolders/pull/3234, everything passes fine)

File: lib/ACL/ACLCacheWrapper.php [hunk 1]
```diff
@@ -15,10 +15,11 @@
 use OCP\Files\Search\ISearchQuery;
 
 class ACLCacheWrapper extends CacheWrapper {
-	private ACLManager $aclManager;
-	private bool $inShare;
+	public function __construct(ICache $cache, private ACLManager $aclManager, private bool $inShare) {
```
[the remark this answers is not in the document]
Comment (provokateurin) [reply lib/ACL/ACLCacheWrapper.php:18]: https://github.com/nextcloud/groupfolders/pull/3249

File: lib/ACL/Rule.php [hunk 2]
```diff
@@ -32,19 +32,14 @@ class Rule implements XmlSerializable, XmlDeserializable, \JsonSerializable {
...
-		$this->fileId = $fileId;
-		$this->mask = $mask;
+	private int $permissions;
+
+	/**
+	 * @param int $mask for every permission type a rule can either allow, deny or inherit
+	 *                  these 3 values are stored as 2 bitmaps, one that masks out all inherit values (1 -> set permission, 0 -> inherit)
+	 *                  and one that specifies the permissions to set for non inherited values (1-> allow, 0 -> deny)
+	 */
+	public function __construct(private IUserMapping $userMapping, private int $fileId, private int $mask, int $permissions) {
```
Comment (come-nc) [inline lib/ACL/Rule.php:42]: break lines (or apply up to date coding standard)

File: lib/ACL/Rule.php [hunk 3]
```diff
@@ -101,7 +90,7 @@ public function xmlSerialize(Writer $writer) {
 	}
 
 	#[\ReturnTypeWillChange]
```
Comment (come-nc) [inline lib/ACL/Rule.php:92]:
```suggestion
```

File: lib/Command/ACL.php [hunk 4]
```diff
@@ -24,25 +24,18 @@
...
-		RuleManager $ruleManager,
-		MountProvider $mountProvider,
-		ACLManagerFactory $aclManagerFactory,
-		IUserManager $userManager
+		protected FolderManager $folderManager,
+		private IRootFolder $rootFolder,
+		private RuleManager $ruleManager,
+		protected MountProvider $mountProvider,
+		private ACLManagerFactory $aclManagerFactory,
+		private IUserManager $userManager
```
Comment (come-nc) [inline lib/Command/ACL.php:33]: Remove promotion for stuff passed to parent constructor

File: lib/Command/Group.php [hunk 5]
```diff
@@ -25,14 +25,12 @@ class Group extends FolderCommand {
 		'share' => Constants::PERMISSION_SHARE,
 		'delete' => Constants::PERMISSION_DELETE,
 	];
-	private IGroupManager $groupManager;
 
-	public function __construct(FolderManager $folderManager, IRootFolder $rootFolder, IGroupManager $groupManager, MountProvider $mountProvider) {
+	public function __construct(protected FolderManager $folderManager, IRootFolder $rootFolder, private IGroupManager $groupManager, MountProvider $mountProvider) {
```
Comment (come-nc) [inline lib/Command/Group.php:29]: same thing

**Review by come-nc — APPROVED** [review]

**Pull request merged**